src/components/SignUpForm/index.jsx

```diff
@@ -1,5 +1,134 @@
 import React from "react";
+import { useForm } from "react-hook-form";
+import Joi from "joi";
+import { joiResolver } from "@hookform/resolvers/joi";
+import InputErrorMessage from "../InputErrorMessage";
+import "./index.scss";
 
-export default function SignUpForm() {
-  return <div>SignUpForm Component</div>;
+const schema = Joi.object({
+  firstName: Joi.string().required().messages({
+    "string.empty": `Please provide your first name`,
+  }),
+  lastName: Joi.string().required().messages({
+    "string.empty": `Please provide your last name`,
+  }),
+  email: Joi.string().required().email({ tlds: {} }).messages({
+    "string.empty": `Please provide your email`,
+    "string.email": `Please provide a valid email`,
+  }),
+  password: Joi.string().required().min(8).max(64).strict().messages({
+    "string.empty": `Please provide a password`,
+  }),
+  confirmPassword: Joi.string()
+    .valid(Joi.ref("password"))
+    .required()
+    .strict()
+    .messages({
+      "any.only": "Those passwords didn't match",
+    }),
+  acceptTerms: Joi.valid("accepted").required().messages({
+    "any.only": `You should accept our terms of service and privcey policy`,
+  }),
+});
+
+export default function SignUpForm({ submit }) {
+  const { register, handleSubmit, errors, reset } = useForm({
+    resolver: joiResolver(schema),
+  });
+
+  const onSubmit = (data) => {
+    submit(data);
+    reset();
+  };
+
+  return (
+    <div className="signUpForm">
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <div className="formInputGroup">
+          <div className="formInput">
+            <input
+              className={errors.email && "inputError"}
+              type="text"
+              placeholder="First name"
+              name="firstName"
+              aria-label="firstName"
+              ref={register()}
+            />
+            <InputErrorMessage errors={errors} name="firstName" />
+          </div>
+
+          <div className="formInput">
+            <input
+              className={errors.email && "inputError"}
```

Comment:
spot the error; this is why you don't copy-paste code

make a function called
getErrorClass

its called as
getErrorClass(errors.email)

Reply:
I see, your notes make me take a step back and re-read my code with Louis's eyes, thank you